pkg/subctl/cmd/cloud/cleanup/rhos.go

```diff
@@ -20,39 +20,49 @@ package cleanup
 
 import (
 	"github.com/spf13/cobra"
-	"github.com/submariner-io/admiral/pkg/reporter"
-	"github.com/submariner-io/cloud-prepare/pkg/api"
 	"github.com/submariner-io/submariner-operator/internal/cli"
 	"github.com/submariner-io/submariner-operator/internal/exit"
+	"github.com/submariner-io/submariner-operator/internal/restconfig"
+	"github.com/submariner-io/submariner-operator/pkg/cloud/cleanup"
+	cloudrhos "github.com/submariner-io/submariner-operator/pkg/cloud/rhos"
+	cmdcloud "github.com/submariner-io/submariner-operator/pkg/subctl/cmd/cloud"
 	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/cloud/rhos"
+	"os"
 )
 
+var rhosConfig cloudrhos.Config
+
 // newRHOSCleanupCommand returns a new cobra.Command used to prepare a cloud infrastructure.
-func newRHOSCleanupCommand() *cobra.Command {
+func newRHOSCleanupCommand(restConfigProducer restconfig.Producer) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "rhos",
 		Short: "Clean up an RHOS cloud",
 		Long: "This command cleans up an OpenShift installer-provisioned infrastructure (IPI) on RHOS-based" +
 			" cloud after Submariner uninstallation.",
-		Run: cleanupRHOS,
-	}
-
-	rhos.AddRHOSFlags(cmd)
+		Run: func(cmd *cobra.Command, args []string) {
+			status := cli.NewReporter()
 
-	return cmd
-}
+			var err error
+			if config.OcpMetadataFile != "" {
+				rhosConfig.InfraID, rhosConfig.ProjectID, err = cloudrhos.ReadFromFile(config.OcpMetadataFile)
+				rhosConfig.Region = os.Getenv("OS_REGION_NAME")
 
-func cleanupRHOS(cmd *cobra.Command, args []string) {
-	err := rhos.RunOnRHOS(*parentRestConfigProducer, "", false, cli.NewReporter(),
-		// nolint:wrapcheck // No need to wrap errors here
-		func(cloud api.Cloud, gwDeployer api.GatewayDeployer, status reporter.Interface) error {
-			err := gwDeployer.Cleanup(status)
-			if err != nil {
-				return err
+				exit.OnErrorWithMessage(err, "Failed to read RHOS Cluster information from OCP metadata file")
+			} else {
+				cmdcloud.ExpectFlag(cmdcloud.InfraIDFlag, rhosConfig.InfraID)
+				cmdcloud.ExpectFlag(cmdcloud.RegionFlag, rhosConfig.Region)
+				cmdcloud.ExpectFlag(cmdcloud.ProjectIDFlag, rhosConfig.ProjectID)
```

Comment:
I want to reuse this section between both `prepare` and `cleanup`. But given the urgency to merge this, it can be refactored in separate commit later on.